client/state/index.js

```diff
@@ -111,7 +114,40 @@ export const reducer = combineReducers( {
 	ui,
 	users,
 	wordads,
-} );
+};
+
+let reducers = { ...initialReducers };
+let onChange = null;
+
+// TODO: See if this export can be eliminated.
+export let reducer = combineReducers( reducers );
+
+export function addReducer( name, reducerFunc ) {
+	if ( reducerExists( name ) ) {
+		throw new Error( 'addReducer(): name "' + name + '" already in use' );
+	}
+
+	reducers = { ...reducers, [ name ]: reducerFunc };
+
+	if ( onChange ) {
+		onChange();
+	}
+}
+
+export function removeReducer( name ) {
+	const { [ name ]: removedReducer, ...remainingReducers } = reducers;
+
+	if ( ! removedReducer ) {
+		log( 'removeReducer(): name not found' );
+		return;
+	}
+
+	reducers = remainingReducers;
```

Comment:
Yeah, I'm not really a fan of this either. I was going for minimal changes.
However, if others agree, I'll add some more here to avoid mutating globals directly like this.